File: spec/models/safe_sfen_spec.rb
require "rails_helper"

RSpec.describe SafeSfen, type: :model do
  it "encode" do
    assert { SafeSfen.encode("position sfen 9/9/9/9/9/9/9/9/9 b - 1") == "cG9zaXRpb24gc2ZlbiA5LzkvOS85LzkvOS85LzkvOSBiIC0gMQ" }
  end
  it "decode" do
    assert { SafeSfen.decode("cG9zaXRpb24gc2ZlbiA5LzkvOS85LzkvOS85LzkvOSBiIC0gMQ") == "position sfen 9/9/9/9/9/9/9/9/9 b - 1" }
  end
end

File: nuxt_side/test/safe_sfen.spec.js
import { SafeSfen } from "@/components/models/safe_sfen.js"

describe("SafeSfen", () => {
  test("encode", () => {
    expect(SafeSfen.encode("position sfen 9/9/9/9/9/9/9/9/9 b - 1")).toEqual("cG9zaXRpb24gc2ZlbiA5LzkvOS85LzkvOS85LzkvOSBiIC0gMQ")
  })

  test("decode", () => {
    expect(SafeSfen.decode("cG9zaXRpb24gc2ZlbiA5LzkvOS85LzkvOS85LzkvOSBiIC0gMQ")).toEqual("position sfen 9/9/9/9/9/9/9/9/9 b - 1")
  })
})
